src/include/sof/audio/component.h

```diff
@@ -755,6 +755,20 @@ static inline void comp_writeback(struct comp_dev *dev)
 		dcache_writeback_region(dev, sizeof(struct comp_dev));
 }
 
+/**
+ * Get component state.
+ *
+ * @param req_dev Requesting component
+ * @param dev Component from which user wants to read status.
+ */
+static inline int comp_get_state(struct comp_dev *req_dev, struct comp_dev *dev)
+{
+	if (req_dev->comp.core != dev->comp.core)
```

Comment:
can we have an inline comment here stating why we invalidate in this case as it's not obvious.

Reply:
@lgirdwood 
I've added inline comment.